### PR TITLE
Cart and search

### PR DESCRIPTION
**Pull request by cecoppinger**
-Added more products to test in the database.  
-Moved the call to getallproducts to the root component so we don't have to call it it in every single other component.  
-The cart is sort of working.  Quantities aren't updated yet but I just wanted to get this in here so that we can use some of the test items in the database for other work.  
-Searching across farmers and products works.  I want to be able to also include product descriptions in search but that's still a work in progress.  To test the search just put in a query into the search bar. 
-I can see that there will be a few conflicts with stephen's pull request.  Looks like it's mostly contained to the productSearchResult component.  Both implementations look like they'll work alright so it probably doesn't matter which one we go with. 

Note:  When the page is refreshed the state is lost which is problematic for the shopping cart.  Is there a way to preserve the state through page refreshes or do we need to add the shopping cart to our backend?

### CODE REVIEW

**Comment by zstephenj** [comment]
Do I need to mess with my database at all? I have the backend up, but I'm still getting a 500 Internal Server Error response from the api. Does that have to do with the SSL?

**Review by zstephenj — APPROVED** [review]

**Comment by iktani** [comment]
same question as I posted on Stephen's PR - are your add/edit product forms working? Can't get either to submit. 
Search works as described. Cart also.

**Review by iktani — APPROVED** [review]

**Review by ryanis-online — APPROVED** [review]

**Comment by ryanis-online** [comment]
I'll merge this once merge conflicts are resolved

**Pull request closed without merging**
